sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -422,6 +432,7 @@ class StreamExecution(
     val replacementMap = AttributeMap(replacements)
     val triggerLogicalPlan = withNewSources transformAllExpressions {
       case a: Attribute if replacementMap.contains(a) => replacementMap(a)
+      case t: CurrentTimestamp => CurrentBatchTimestamp(currentBatchTimestamp)
```

Comment:
@hvanhovell Timestamp is "non-deterministic" when it comes to streaming aggregations. Therefore it shouldn't be pushed down stateful aggregations. Otherwise the resulting table you get is different than what you asked for.
Example: 
```
df.groupBy('time).agg(count("*")).where('time >= current_timestamp().cast("long") - 30 * 60)
```
What you get:
All aggregates from all of time, but the aggregates older than 30 minutes don't get updated. Those get filtered
What I wanted:
Just return me the last 30 minutes of aggregates. I don't care about anything else older.